File: notebooks/breastcancer with r/model_training.r
library(tidymodels)
library(xgboost)

max_depth <- as.integer(Sys.getenv('max_depth'))
n_round <- as.integer(Sys.getenv('n_round'))
early_stopping_rounds <- as.integer(Sys.getenv('early_stopping_rounds'))


print('Loading training and test data')
train <- read.csv("train.csv", header = TRUE)
test <- read.csv("test.csv", header = TRUE)

#Remove labels 
train_data <- train %>% select(-diagnosis)
test_data <- test %>% select(-diagnosis)

#create targets

train_label <- train %>%
  select(diagnosis) %>%
  mutate(diagnosis = ifelse(diagnosis == "M", 1, 0)) %>%
  pull(diagnosis)

test_label <- test %>%
  select(diagnosis) %>%
  mutate(diagnosis = ifelse(diagnosis == "M", 1, 0)) %>%
  pull(diagnosis)

#convert to matrices
train_matrix <- as.matrix(train_data)
train_label <- as.numeric(train_label)

test_matrix <- as.matrix(test_data)
test_label <- as.numeric(test_label)


#Create DMatrix
dtrain <- xgb.DMatrix(data = train_matrix, label = train_label)
dtest <- xgb.DMatrix(data = test_matrix, label = test_label)


print('Training model...')
breastcancer_model <-xgboost(data = dtrain, 
                              nround = n_round,
                              max.depth = max_depth,
                              early_stopping_rounds = early_stopping_rounds,
                              verbose = 2,
                              objective = "binary:logistic")

pred <- predict(breastcancer_model, dtest)
err <- mean(as.numeric(pred > 0.5) != test_label)
print(paste("test-error=", err))

print('Saving model')
xgb.save(breastcancer_model, 'model.bst') 

File: notebooks/breastcancer prediction with r and python/python/model_training.py
from os import environ

from numpy import mean
from pandas import read_csv
from xgboost import DMatrix, train


# Read environment variables
max_depth = int(environ.get('max_depth', '10'))
n_round = int(environ.get('n_round', '21'))
early_stopping_rounds = int(environ.get('early_stopping_rounds', '3'))

print('Loading training and test data')
train_set = read_csv("train.csv")
test_set = read_csv("test.csv")

# Remove labels
train_data = train_set.drop(columns=['diagnosis'])
test_data = test_set.drop(columns=['diagnosis'])

# Create targets
train_label = train_set['diagnosis'].apply(lambda x: 1 if x == "M" else 0).values
test_label = test_set['diagnosis'].apply(lambda x: 1 if x == "M" else 0).values

# Convert to matrices
train_matrix = train_data.values
test_matrix = test_data.values

# Create DMatrix
dtrain = DMatrix(data=train_matrix, label=train_label)
dtest = DMatrix(data=test_matrix, label=test_label)

print('Training model...')
breastcancer_model = train(
    params={
        'max_depth': max_depth,
        'objective': 'binary:logistic',
        'verbosity': 2,
        'early_stopping_rounds': early_stopping_rounds
    },
    dtrain=dtrain,
    num_boost_round=n_round,
    evals=[(dtest, 'test')]
)

# Predict and evaluate
pred = breastcancer_model.predict(dtest)
err = mean((pred > 0.5).astype(int) != test_label)
print(f"test-error= {err}")

print('Saving model')
breastcancer_model.save_model('model.bst')